src/ClientBuilder.php

```diff
@@ -136,60 +143,35 @@ public function setSdkVersion(string $sdkVersion): ClientBuilderInterface
         return $this;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function setTransportFactory(TransportFactoryInterface $transportFactory): ClientBuilderInterface
+    public function getTransport(): TransportInterface
     {
-        $this->transportFactory = $transportFactory;
-
-        return $this;
+        return $this->transport;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getClient(): ClientInterface
+    public function setTransport(TransportInterface $transport): ClientBuilderInterface
     {
-        $this->transport = $this->transport ?? $this->createTransportInstance();
+        $this->transport = $transport;
 
-        return new Client($this->options, $this->transport, $this->sdkIdentifier, $this->sdkVersion, $this->serializer, $this->representationSerializer, $this->logger);
+        return $this;
     }
 
-    /**
-     * Creates a new instance of the transport mechanism.
-     */
-    private function createTransportInstance(): TransportInterface
+    public function getHttpClient(): HttpClientInterface
     {
-        if (null !== $this->transport) {
-            return $this->transport;
-        }
+        return $this->httpClient;
+    }
 
-        $transportFactory = $this->transportFactory ?? $this->createDefaultTransportFactory();
+    public function setHttpClient(HttpClientInterface $httpClient): ClientBuilderInterface
+    {
+        $this->httpClient = $httpClient;
 
-        return $transportFactory->create($this->options);
+        return $this;
     }
 
     /**
-     * Creates a new instance of the {@see DefaultTransportFactory} factory.
+     * {@inheritdoc}
      */
-    private function createDefaultTransportFactory(): DefaultTransportFactory
+    public function getClient(): ClientInterface
```

Comment:
Having `getClient` and `getHttpClient` may be confusing... What's the difference? Can we use another name?

We also have Transport too!

Reply:
Should we mark that as `@internal` then?

Reply:
Totally, this is all still WIP, as mentioned in the PR description. 😃
Docs and annotation will follow once the implementation and tests are in place.